hw/ip/flash_ctrl/data/flash_ctrl_testplan.hjson

```diff
@@ -7,21 +7,212 @@
   entries: [
     {
       name: sanity
       desc: '''
-            Basic sanity test acessing a major datapath within the flash_ctrl.
-
-            **Stimulus**:
-            - TBD
-
-            **Checks**:
-            - TBD
+            Randomly read, program or erase (page or a bank) a randomized chunk of flash memory.
+            Only the data partition is accessed. No extra features enabled. Flash memory is
+            invalidated and the targeted chunk is initialized with random data for reads and all 1s
+            for writes. Interrupts are not enabled, Completion is ascertained through polling. The
+            success of each operation is verified via backdoor.
             '''
       milestone: V1
       tests: ["flash_ctrl_sanity"]
     }
+    {
+      name: backpressure
+      desc: '''
+            Sanity + randomly pick between back-pressured reads / writes and fifo status-polled
+            reads / writes via the controller.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: all_partitions
+      desc: '''
+            Sanity + both, legal data and info partitions are accessed. In future, support for
+            multiple info partitions may be added - those will be covered as well.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: host_read_check
+      desc: '''
+            Sanity + the success of each operation is verified by randomly choosing
+            between backdoor and via the host interface.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: host_illegal_access
+      desc: '''
+            Do a bunch of illegal accesses via host interface to ensure that an error response is
+            appropriately seen. This includes writes, out of bound reads, protocol violated reads
+            and all other unsupported accesses.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: host_outstanding_access
+      desc: '''
+            Drive back-to-back access to ensure coverage on the maximum supported outstanding
+            on this interface.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: intr_enabled
+      desc: '''
+            Sanity + enable all interrupts and verify correctness of assertion and de-assertion of
+            each pin. Replace polling with waits on interrrupts.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: host_ctrl_arb
+      desc: '''
+            Continuously read data from host interface while randomly programming or reading from
+            the flash controller. Ensure the expected bandwidths are seen on both interfaces.
+            '''
+      milestone: V2
+      tests: []
+    }
+    {
+      name: host_read_collide
+      desc: '''
+            Read LSB bus word via host and program or erase the MSB bus word at the same time. Read
```

Comment:
This one is really good, but we may need to be a bit explicit about which one wins arbitration. 

For example let's say the host read wins, then that means the host would read back the pre-updated data.
If the program / erase wins, then the read would get the updated data.

Reply:
I updated this a little. Since erase operates on full page granularity, it would be better to either program or read the MSB word instead. Regardless of which one wins, the second read via host should always read back the updated data, agree?

Reply:
just to confirm, when you say second read, you mean first read / program collide (one of these will finish first, the other second), then ANOTHER following read should get the updated data? 

Or do you mean that as long as the read goes second during a collision, it should always get the updated data?

Reply:
First, a read via host and read/program via the controller are made at the same time to two different bus words (but belonging to the same flash data word). This creates a collision. Either one can finish first. Regardless of which one does, both operations should succeed. 

Then, a second read is made via host, this time with no collision, but to the bus word we programmed in the first step. If the internal buffer was cleared properly, then we should see the updated result. 

Does this sound good?